server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -379,7 +379,7 @@ static boolean executeBulkItemRequest(
             } catch (Exception e) {
                 logger.info(() -> format("%s mapping update rejected by primary", primary.shardId()), e);
                 assert result.getId() != null;
-                onComplete(exceptionToResult(e, primary, isDelete, version, result.getId()), context, updateResult);
```

Comment:
Do you mean `dynamic: true_until_limit` instead of `ignore_above`? I'm not sure how to access the `dynamic` flag from this point. I think that flag may also be set in some nested object instead of at the root level of the mappings so it won't be trivial to determine if `true_until_limit` has been set anywhere in the mapping.

Reply:
I'm now only retrying if the mapping version has changed.